jcl/src/java.base/share/classes/com/ibm/jit/JITHelpers.java

```diff
@@ -135,6 +135,37 @@ private static int classIsInterfaceFlag() {
 	}
 
 	/*
+	 * To be recognized by the JIT and turned into trees that is foldable when obj is known at compile time.
+	 */
+	public boolean isArray(Object obj) {
+		if (is32Bit()) {
+			int j9Class = getJ9ClassFromObject32(obj);
+			int flags = getClassDepthAndFlagsFromJ9Class32(j9Class);
+			return (flags & VM.J9_ACC_CLASS_ARRAY) == 0 ? false : true;
```

Comment:
Is there a specific reason you need the ? here (for the JIT)?  This would read better as simply != 0.

Reply:
Fixed.